tests/unit/test_config_graph_sync.py

```
@@ -0,0 +1,67 @@
+    with pytest.raises(ValidationError, match="graph_sync_mode='async_outbox' requires"):
+        _make_settings(graph_enabled=False, graph_sync_mode="async_outbox")
+
+
+def test_supabase_with_graph_requires_async_outbox() -> None:
+    with pytest.raises(ValidationError, match="Supabase \\+ graph"):
+        _make_settings(
+            storage_backend="supabase",
+            supabase_url="https://example.supabase.co",
+            supabase_key="key",
+            graph_enabled=True,
+            neo4j_password="secret",
```

Ruff (0.15.14)

[error] Possible hardcoded password assigned to argument: "neo4j_password"

(S106)